### PR TITLE
Handle DNS queries of type MX

### PR DESCRIPTION
**Pull request by ddebroy**
The Docker embedded server was not handling DNS queries of type MX for containers. They were instead being passed down to the host leading to `NXDOMAIN` responses not compliant with SMTP RFC 5321 (in case someone wanted to host a mail server in a container and deliver emails to the server). This PR adds support to respond to MX queries in a way compliant with RFC 5321 for containers.

Note that providing a full mechanism for starting a mail server in a container/service and registering MX RRs within the Docker resolver is beyond the scope of this PR.

### CODE REVIEW

**Comment by ddebroy** [comment]
Addresses https://github.com/docker/libnetwork/issues/2026

**Comment by thaJeztah** [comment]
Do we need a unit-test for this?

**Comment by ddebroy** [comment]
@thaJeztah good point. I was indeed looking for unit-tests around the resolver to add something for this but could not find anything. If there is a suite already, let me know. Otherwise we will need something like `resolver_test.go` to cover all the name query scenarios.

File: resolver.go [hunk 1]
```diff
@@ -224,6 +224,22 @@ func createRespMsg(query *dns.Msg) *dns.Msg {
...
 
+func (r *resolver) handleMXQuery(name string, query *dns.Msg) (*dns.Msg, error) {
+	addrv4, _ := r.backend.ResolveName(name, types.IPv4)
+	addrv6, _ := r.backend.ResolveName(name, types.IPv6)
+
+	if addrv4 == nil && addrv6 == nil {
+		return nil, nil
+	}
+
+	// We were able to resolve the name. Respond with an empty list with
```
Comment (fcrisciani) [inline resolver.go]: I see this comment but not sure how is it implemented below? does it mean that is enough to just reply with the same records?

Comment (ddebroy) [reply resolver.go]: If we are able to resolve the hostname, for MX, we need to reply with a DNS msg with empty answers section (with status set to default RcodeSuccess). `createRespMsg` generates the DNS msg with empty answers section and status set to RcodeSuccess. This is very similar to what we are doing down below for ipv6 AAAA querries that resolve to a ipv4 address.

**Comment by ddebroy** [comment]
I have added the unit test for resolver.

**Comment by selansen** [comment]
LGTM

**Pull request merged**